server/src/main/java/org/apache/kafka/server/AssignmentsManager.java

```diff
@@ -95,7 +96,10 @@ public void close() throws InterruptedException {
     }
 
     public void onAssignment(TopicIdPartition topicPartition, Uuid dirId) {
```

Comment:
Do we need two constructors for this? Seems like someone could just pass a callback that doesn't do anything, if they don't want any.

Also, I don't see a null check later for `assignmentEvent.callback` ... so is null even allowed here?

Reply:
I changed this to have one `onAssignment` and `AssignmentEvent` constructor that will replace callback null with runnable that does nothing

Reply:
Should we do the same to `DirectoryEventHandler.handleAssignment` for consistency?